quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/LoggingPropertyMappers.java

```diff
@@ -387,16 +420,24 @@ private static String resolveLogOutput(String value, ConfigSourceInterceptorCont
     /**
      * Add tracing info to the log if the format is not explicitly set, and tracing and {@code includeTraceOption} options are enabled
      */
-    private static String addTracingInfo(String value, Option<Boolean> includeTraceOption) {
+    private static String addTracingAndMdcInfo(String value, Option<Boolean> includeTraceOption, Option<Boolean> includeMdcOption) {
         var isTracingEnabled = TracingPropertyMappers.isTracingEnabled();
         var includeTrace = isTrue(includeTraceOption);
         var isChangedLogFormat = !DEFAULT_LOG_FORMAT.equals(value);
+        var includeMdc = isTrue(includeMdcOption);
 
-        if (!isTracingEnabled || !includeTrace || isChangedLogFormat) {
+        if (isChangedLogFormat) {
             return value;
         }
 
-        return LoggingOptions.DEFAULT_LOG_TRACING_FORMAT;
+        StringBuilder additionalFields = new StringBuilder();
+        if (isMdcActive() && includeMdc) {
+            additionalFields.append("%X ");
+        } else if (isTracingEnabled && includeTrace) {
+            additionalFields.append("traceId=%X{traceId}, parentId=%X{parentId}, spanId=%X{spanId}, sampled=%X{sampled} ");
+        }
```

Comment:
It breaks the log format and order of the trace info. I want to double-check if it is ok.

26.2 `2025-06-23 19:26:51,218 traceId=, parentId=, spanId=, sampled= INFO  [io.quarkus] ...`
This PR: `2025-06-23 19:25:02,065 {} INFO [io.quarkus]`

26.2: `2025-06-23 19:20:32,847 traceId=f2a0091391959c796465cec6f835b9f6, parentId=43eb89c4fd4ef91e, spanId=a2afc597b272292c, sampled=true INFO  [org.keycloak.services] ...`
This PR: `2025-06-23 19:25:22,781 {kc.realm=master, parentId=dfa43ccbb03ab7d2, sampled=true, spanId=4f87ba42d9b77505, traceId=b90db8ca3fc9031947124946cf8535e6} INFO  [org.keycloak.services]  ...`

Reply:
I would think that the new `%X` is superior, and I would suggest to eventually use that everywhere, maybe in KC 27. I would then update the docs accordingly. Even if neither MDC nor tracing is enabled, I would always want to go with `%X` which would then print a pair of curly braces. It removes variance.